lexicon/providers/glesys.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from .base import Provider as BaseProvider
+import requests
+import json
+
+def ProviderParser(subparser):
+    subparser.add_argument("--auth-username", help="specify username (CL12345)")
+    subparser.add_argument("--auth-token", help="specify API key")
+
+class Provider(BaseProvider):
+
+    def __init__(self, options, provider_options={}):
+        super(Provider, self).__init__(options)
+        self.domain_id = None
+        self.api_endpoint = provider_options.get('api_endpoint') or 'https://api.glesys.com'
+
+    # Provider does not support this. Uses basic auth.
+    def authenticate(self):
+        pass
```

Comment:
The authenticate method is primarily used to validate that the domain whose DNS records we're going to update/modify is actually owned by the user. Something like the https://github.com/glesys/API/wiki/functions_domain#domainlist endpointshould be used.

The idea is to fail quickly if something is wrong.